api/src/main/java/com/spectralogic/tpfr/api/ServerServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+    public IndexStatusResponse indexFile(final String filePath) throws IOException {
+        final Response<IndexStatusResponse> response = api.indexFile(filePath).execute();
+
+        if (!response.isSuccessful()) {
+            LOG.error("indexFile api call failed ({}, {})", response.code(), response.message());
+            return new IndexStatusResponse("Unknown", null, null, null,
+                    null, String.valueOf(response.code()), response.message());
+        }
+
+        return response.body();
+    }
+
+    @Override
+    public IndexStatusResponse fileStatus(final String filePath) throws IOException {
+        final Response<IndexStatusResponse> response = api.fileStatus(filePath).execute();
+
+        if (!response.isSuccessful()) {
+            LOG.error("fileStatus api call failed ({}, {})", response.code(), response.message());
+            return new IndexStatusResponse("Unknown", null, null, null,
+                    null, String.valueOf(response.code()), response.message());
+        }
+
+        return response.body();
+    }
+
+    @Override
+    public OffsetsStatusResponse questionTimecode(final Map<String, String> params) throws IOException {
+        final Response<OffsetsStatusResponse> response = api.questionTimecode(params).execute();
+
+        if (!response.isSuccessful()) {
+            LOG.error("questionTimecode failed ({}, {})", response.code(), response.message());
+            return new OffsetsStatusResponse("Unknown", null, null);
+        }
+
+        return response.body();
+    }
+
+    @Override
+    public ReWrapResponse reWrap(final Map<String, String> params) throws IOException {
+        final Response<ReWrapResponse> response = api.reWrap(params).execute();
+
+        if (!response.isSuccessful()) {
+            // Due to bad design in Marquis
+            if (response.code() == 400 && response.message().equals("OK")) {
+                return getErrorBody(response.errorBody(), ReWrapResponse.class);
+            }
+            LOG.error("reWrap failed ({}, {})", response.code(), response.message());
+            return new ReWrapResponse("Unknown");
+        }
+
+        return response.body();
+    }
+
+    @Override
+    public ReWrapStatusResponse reWrapStatus(final String targetFileName) throws IOException {
+        final Response<ReWrapStatusResponse> response = api.reWrapStatus(targetFileName).execute();
+
+        if (!response.isSuccessful()) {
+            LOG.error("reWrapStatus failed ({}, {})", response.code(), response.message());
+            return new ReWrapStatusResponse("Unknown", null, null,
+                    String.valueOf(response.code()), response.message());
+        }
+
+        return response.body();
+    }
+
+    private <T> T getErrorBody(final ResponseBody responseBody, final Class<T> clazz) throws IOException {
+        final Converter<ResponseBody, T> errorConverter =
+                retrofit.responseBodyConverter(clazz, new Annotation[0]);
```

Comment:
How hard would it be to get the  Converter directly rather than from retrofit?  IE, can we not pass in the retrofit instance but still get what we need to parse the error message.